gluefactory/datasets/merge_datasets.py

```diff
@@ -60,11 +66,26 @@ def __init__(self, conf, split):
         #self.set_split_for_all_datasets(split)
         self.datasets = {}  # store dataset objects
         self.img_index_collection = []  # store image indices
+        if self.conf.use_multiscale_learning:
+            self.relevant_batch_size = self.conf[f"{split}_batch_size"]
+            self.num_selected_this_batch = 0
```

Comment:
What does num_selected_this_batch mean?

Reply:
This variable represents the number of selected images that are already selected for the current batch.
It is needed to synchronize multi-scale-training between two datasets that are merged. Because every time we are switching to a new batch we potentially switch scales.

If needed I can also explain the full procedure. Lmk